pywinauto/macos/macos_functions.py

```diff
@@ -0,0 +1,236 @@
+    return NSScreen.mainScreen().frame()
+
+def read_from_clipboard():
+    return subprocess.check_output('pbpaste', env={'LANG': 'en_US.UTF-8'}).decode('utf-8')
+
+def get_windows_list_info():
+    list_info = CGWindowListCopyWindowInfo(kCGWindowListOptionOnScreenOnly, kCGNullWindowID)
+    if list_info is not None:
+        return list_info
+
+def print_list_of_atrributes(ax_element):
+    for iter in dir(ax_element):
+        print (iter)
+
+def check_is_process_trusted():
+    if not (AXIsProcessTrusted()):
+        if (is_debug):
+            print("Process is not Trusted")
+        exit(-1)
+
+def from_py_objc_to_apple_attribute(pyObjcName):
+    return six.text_type(pyObjcName)
+
+def get_list_of_attributes(ax_element):
+    list_of_options_result_with_error_code = AXUIElementCopyAttributeNames(ax_element,None)
+    if not (check_error(list_of_options_result_with_error_code)):
+        list_of_options = list_of_options_result_with_error_code[1]
+        return list_of_options
+
+def check_error(obj):
+    if obj[0] == kAXErrorSuccess:
+        return False
+    else:
+        if (is_debug):
+            print("Capture the error with code"+str(obj[0]))
+        return True
+
+def check_attribute_valid(ax_element,attribute):
+    list_of_options = get_list_of_attributes(ax_element)
+    if list_of_options is None:
+        return False
+    for iter in list_of_options:
+        if iter == attribute:
+            return True
+    return False
+
+def get_ax_attribute(ax_element,attribute_name):
+    if check_attribute_valid(ax_element,attribute_name):
+        new_attribute_name = from_py_objc_to_apple_attribute(attribute_name)
+        fetch_result_with_error = AXUIElementCopyAttributeValue(ax_element,unicode(attribute_name),None)
+        if not (check_error(fetch_result_with_error)):
+            fetchResult = fetch_result_with_error[1]
+            return fetchResult
+        if (is_debug):
+            print("Attribute " + new_attribute_name + " is not defined" )
+
+def get_list_of_actions(ax_element):
+    list_of_options_result_with_error_code = AXUIElementCopyActionNames(ax_element,None)
+    if not (check_error(list_of_options_result_with_error_code)):
+        list_of_options = list_of_options_result_with_error_code[1]
+        return list_of_options
+
+def perform_action(ax_element,action):
+    AXUIElementPerformAction(ax_element,action)
+
+def print_child_tree(ui_element_ref,count = 0):
+    role = get_ax_attribute(ui_element_ref,"AXRole")
+    if role is not None:
+        print("-"*count + role)
+    childrens = get_ax_attribute(ui_element_ref,"AXChildren")
+    if childrens is not None:
+        for item in childrens:
+            if item is None:
+                continue
+            print_child_tree(item,count + 1)
+
+def get_descendants(root,descendants):
+    childrens = get_ax_attribute(ui_element_ref,"AXChildren")
+    if childrens is not None:
+        for child in childrens:
+            if child is None:
```

Comment:
Looks like it must be quite opposite:
`if child is not None:` or just `if child:`